pkg/sql/catalog/descs/uncommitted_descriptors.go

```diff
@@ -186,7 +187,16 @@ func (ud *uncommittedDescriptors) upsert(
 		newBytes += mut.ByteSize()
 	}
 	ud.mutable.Upsert(mut)
-	uncommitted := mut.ImmutableCopy()
+	// Upsert an immutable copy of `mut` with its raw bytes in storage
+	// updated to itself. The raw bytes serves as the expected bytes to
+	// `CPut` when `WriteDescToBatch` updates this descriptor again
+	// in the future.
+	var val roachpb.Value
+	val.SetProto(mut.DescriptorProto())
+	rawBytesInStorage := val.TagAndDataBytes()
```

Comment:
I'm sorry to say that this won't work. Protobuf serialization and deserialization is not bijective so you can't infer what the bytes in storage look like. 

A concrete example is when we remove certain fields in the descriptor proto definition: roundtripping like this will lose the bytes corresponding to the removed fields. For this to happen in a cluster you have to have a descriptor that hasn't changed over two major versions; which isn't as rare as you'd think.

Reply:
Turns out I got it wrong a bit, we discussed this offline.